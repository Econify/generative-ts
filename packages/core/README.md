# @generative-ts/core

Core functionality for generative-ts, see [main README](https://github.com/Econify/generative-ts/blob/main/README.md)
