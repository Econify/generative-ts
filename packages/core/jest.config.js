const { pathsToModuleNameMapper } = require('ts-jest');
const { compilerOptions } = require('../../tsconfig.base.json');

module.exports = {
  displayName: 'core',
  testEnvironment: 'node',
  rootDir: '../../',
  roots: ['<rootDir>/packages/core/src'],
  transform: {
    '^.+\\.ts$': 'ts-jest'
  },
  testMatch: ['<rootDir>/packages/core/src/**/*.spec.ts'],
  moduleFileExtensions: ['ts', 'js', 'json', 'node'],
  moduleNameMapper: pathsToModuleNameMapper(compilerOptions.paths, { prefix: '<rootDir>/' })
};
