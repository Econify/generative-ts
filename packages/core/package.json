{
  "name": "@generative-ts/core",
  "version": "0.1.0-alpha.3",
  "description": "Core functionalities for generative-ts",
  "main": "dist/bundle.cjs.js",
  "module": "dist/bundle.esm.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "npm run clean && rollup -c",
    "check": "npm run lint && npm run test",
    "clean": "rm -rf dist",
    "lint": "eslint src/",
    "lint:fix": "eslint --fix src/",
    "prepublishOnly": "npm run check && npm run build",
    "test": "jest",
    "test:cov": "jest --coverage"
  },
  "dependencies": {
    "aws4": "^1.12.0",
    "fp-ts": "^2.16.5",
    "io-ts": "^2.2.21",
    "nunjucks": "^3.2.4",
    "tslib": "^2.6.2"
  },
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Econify/generative-ts.git"
  },
  "author": "John M Naglick",
  "license": "ISC",
  "bugs": {
    "url": "https://github.com/Econify/generative-ts/issues"
  },
  "homepage": "https://github.com/Econify/generative-ts#readme",
  "keywords": [
    "llm",
    "gpt",
    "llama",
    "ts",
    "generative",
    "typescript"
  ]
}
