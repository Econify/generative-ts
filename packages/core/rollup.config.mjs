import { baseConfig } from '../../rollup.config.base.mjs';

// import pkg from './package.json' assert { type: 'json' };
const pkg = { name: '@generative-ts/core' };

export default [
  {
    ...baseConfig,
    input: 'src/index.ts',
    output: [
      {
        file: 'dist/index.cjs',
        format: 'cjs',
        sourcemap: true
      },
      {
        file: 'dist/index.mjs',
        format: 'esm',
        sourcemap: true
      }
    ],
    plugins: [
      ...baseConfig.plugins,
      {
        name: 'finished',
        buildStart() {
          console.log(`\x1b[32mBundling ${pkg.name}...\x1b[0m`);
        },
        writeBundle({ format }) {
          console.log(`\x1b[32mFinished \x1b[1m${pkg.name}\x1b[0m (${format})`);
        }
      }
    ],
    external: [
      ...baseConfig.external,
      'aws4',
      'ejs',
      'fp-ts/lib/Either.js',
      'io-ts',
    ]
  }
];
