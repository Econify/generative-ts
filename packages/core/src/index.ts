/*
 * Public API - APIs
 */
export {
  Ai21Jurassic2Api,
  Ai21Jurassic2Options,
  Ai21Jurassic2Response,
  Ai21Jurassic2Template,
} from "./apis/ai21";
export {
  AmazonTitanTextApi,
  AmazonTitanTextOptions,
  AmazonTitanTextResponse,
  AmazonTitanTextTemplate,
} from "./apis/amazon";
export {
  CohereChatApi,
  CohereChatOptions,
  CohereChatResponse,
  CohereChatTemplate,
  CohereGenerateApi,
  CohereGenerateOptions,
  CohereGenerateResponse,
  CohereGenerateTemplate,
} from "./apis/cohere";
export {
  GoogleGeminiApi,
  GoogleGeminiOptions,
  GoogleGeminiResponse,
  GoogleGeminiTemplate,
} from "./apis/google";
export {
  HfConversationalTaskApi,
  HfConversationalTaskOptions,
  HfConversationalTaskResponse,
  HfConversationalTaskTemplate,
  HfInferenceApiOptions,
  HfTextGenerationTaskApi,
  HfTextGenerationTaskOptions,
  HfTextGenerationTaskResponse,
  HfTextGenerationTaskTemplate,
} from "./apis/huggingface";
export {
  LlamaResponse,
  Llama2ChatApi,
  Llama2ChatOptions,
  Llama2ChatTemplate,
  Llama3ChatApi,
  Llama3ChatOptions,
  Llama3ChatTemplate,
} from "./apis/meta";
export {
  MistralAiApi,
  MistralAiOptions,
  MistralAiResponse,
  MistralAiTemplate,
  MistralBedrockApi,
  MistralBedrockOptions,
  MistralBedrockResponse,
  MistralBedrockTemplate,
} from "./apis/mistral";
export {
  OpenAiChatApi,
  OpenAiChatOptions,
  OpenAiChatResponse,
  OpenAiChatTemplate,
} from "./apis/openai";

/*
 * Public API - Providers
 */
export {
  BaseModelProvider,
  HttpModelProvider,
  createAwsBedrockModelProvider,
  createCohereModelProvider,
  createGroqModelProvider,
  createHuggingfaceInferenceModelProvider,
  createLmStudioModelProvider,
  createMistralModelProvider,
  createOpenAiChatModelProvider,
  AwsAuthConfig,
  CohereAuthConfig,
  GroqAuthConfig,
  HuggingfaceAuthConfig,
  MistralAuthConfig,
  OpenAiAuthConfig,
} from "./providers";

/*
 * Public API - Utils
 */
export { EjsTemplate } from "./utils";

/*
 * Public API - TypeDefs
 */
export {
  Template,
  ModelId,
  ModelRequestOptions,
  ModelApi,
  ModelProvider,
  HttpClient,
  HttpClientRequest,
} from "./typeDefs";
