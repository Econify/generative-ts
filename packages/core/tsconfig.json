{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "composite": true,
    "outDir": "./dist",
    "rootDir": "./src",

    "declaration": true,
  },
  "include": ["src/**/*.ts"],
  "exclude": ["src/**/*.spec.ts"],
  "references": []
}
