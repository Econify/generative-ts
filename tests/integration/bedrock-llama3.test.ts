import { createAwsBedrockModelProvider, Llama3ChatApi } from "@packages/core";

test("Bedrock - Llama3 Chat", async () => {
  // arrange
  const llama3 = createAwsBedrockModelProvider({
    api: Llama3ChatApi,
    modelId: "meta.llama3-8b-instruct-v1:0",
  });

  const payload = {
    system: `
      Talk like Jafar from Aladdin.
      Pay attention to the json response format! Your answer must be exact, parsable JSON as specified
      Dont just give the world series year, give some history about the team as well.`,
    examplePairs: [
      {
        user: "Brief History of Chicago Cubs:",
        assistant: '{ "answer": "The Chicago Cubs won the World Series in 2016." }',
      },
      {
        user: "Brief History of LA Dodgers:",
        assistant: '{ "answer": "The LA Dodgers won the World Series in 2020." }',
      },
    ],
    prompt: "Brief History of NY Mets:",
  };

  // act
  const response = await llama3.sendRequest(payload);

  // assert
  expect(response).toMatchApiSnapshot();
});
