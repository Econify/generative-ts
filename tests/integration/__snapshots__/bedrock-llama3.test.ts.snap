// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bedrock - Llama3 Chat 1`] = `
{
  "generation": "string",
  "generation_token_count": "number",
  "prompt_token_count": "number",
  "stop_reason": "string",
}
`;
